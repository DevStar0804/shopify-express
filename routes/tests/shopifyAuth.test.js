const findFreePort = require('find-free-port')
const fetch = require('node-fetch');
const http = require('http');
const express = require('express');

const { MemoryStrategy } = require('../../strategies');
const createShopifyAuthRouter = require('../shopifyAuth');

const PORT = 3000;
const BASE_URL = `http://localhost:${PORT}`

let server;
let afterAuthSpy;
describe('shopifyAuth', async () => {
  beforeEach(async () => {
    afterAuthSpy = jest.fn();

    server = await createServer(afterAuthSpy);
  });

  afterEach(() => {
    server.close();
  });

  describe('/', () => {
    it('responds to get requests by returning a redirect page', async () => {
      const response = await fetch(`${BASE_URL}/?shop=shop1`);
      const data = await response.text();

      expect(response.status).toBe(200);
      expect(data).toMatchSnapshot();
    });

    it('responds with a 400 when no shop query parameter is given', async () => {
      const response = await fetch(BASE_URL);
      const data = await response.text();

      expect(response.status).toBe(400);
      expect(data).toMatchSnapshot();
    });
  });

  describe('/callback', () => {
    it('errors when hmac validation fails', () => {
      pending();
    });

    it('does not error when hmac validation succeds', () => {
      pending();
    });

    it('requests access token', () => {
      pending();
    });

    it('console warns when no session is present on request context', () => {
      pending();
    });
  });
});

function createServer(afterAuth) {
  const app = express();

  app.use(
    '/',
    createShopifyAuthRouter({
      apiKey: 'key',
      secret: 'secret',
      scope: ['scope'],
      shopStore: new MemoryStrategy(),
      afterAuth,
    }),
  );

  server = http.createServer(app);

  return new Promise((resolve, reject) => {
    findFreePort(PORT, (err, freePort) => {
      if (err) {
        throw err;
      }
      server.listen(PORT, resolve(server));
    })
  });
}
