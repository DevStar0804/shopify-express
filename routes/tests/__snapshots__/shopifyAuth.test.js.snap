// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shopifyAuth / responds to get requests by returning a redirect page 1`] = `
"<!DOCTYPE html>
      <html>
        <head>
          <script type=\\"text/javascript\\">
            window.top.location.href = \\"https://shop1/admin/oauth/authorize?client_id=key&scope=scope&redirect_uri=undefined/auth/shopify/callback\\"
          </script>
        </head>
      </html>"
`;

exports[`shopifyAuth / responds with a 400 when no shop query parameter is given 1`] = `"Expected a shop query parameter"`;
